lib/llvmopencl/TargetAddressSpaces.cc

```diff
@@ -92,6 +92,9 @@ TargetAddressSpaces::runOnModule(llvm::Module &M) {
 
   llvm::StringRef arch(M.getTargetTriple());
 
+  if (arch.startswith("spir"))
+    arch = OCL_KERNEL_TARGET;
```

Comment:
Yes, I meant heterogeneous platforms with one or more non-host device. 

For the codegen, check 'basic.c', scan for llvm_codegen() (it's in common.c). But the kernel compilation passes (that work on LLVM IR) are driven from pocl_llvm_api.cc. It _might_ work by just forcing the real target machine be created in GetTargetMachine() instead of the SPIR target and fetching the actual target machine (instead of the module triplet) in TargetAddressSpaces.